src/pages/ReportAvatar.tsx

```diff
@@ -21,23 +22,53 @@ type ReportAvatarOnyxProps = {
 type ReportAvatarProps = ReportAvatarOnyxProps & StackScreenProps<AuthScreensParamList, typeof SCREENS.REPORT_AVATAR>;
 
 function ReportAvatar({report = {} as Report, route, policies, isLoadingApp = true}: ReportAvatarProps) {
+    const {translate} = useLocalize();
     const policyID = route.params.policyID ?? '-1';
     const policy = policies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
-    const policyName = ReportUtils.getPolicyName(report, false, policy);
-    const avatarURL = ReportUtils.getWorkspaceAvatar(report);
+    const policyName = ReportUtils.getPolicyName(report, true, policy);
+    const isPolicyRelatedReport = !!policyName;
+    const unavailableWorkspace = translate('workspace.common.unavailable');
+
+    const attachment = useMemo(() => {
+        if (isPolicyRelatedReport) {
+            return {
+                source: UserUtils.getFullSizeAvatar(ReportUtils.getWorkspaceAvatar(report), 0),
+                headerTitle: policyName,
+                originalFileName: policy?.originalFileName ?? policy?.id ?? report?.policyID ?? '',
+                isWorkspaceAvatar: true,
+            };
+        }
+
+        if (ReportUtils.isGroupChat(report) && !ReportUtils.isThread(report)) {
+            return {
+                source: report?.avatarUrl ? UserUtils.getFullSizeAvatar(report.avatarUrl, 0) : ReportUtils.getDefaultGroupAvatar(report?.reportID ?? ''),
+                headerTitle: ReportUtils.getReportName(report),
+                originalFileName: report?.avatarFileName ?? '',
+                isWorkspaceAvatar: false,
+            };
+        }
+
+        return {
+            source: UserUtils.getFullSizeAvatar(ReportUtils.getWorkspaceAvatar(report), 0),
+            headerTitle: unavailableWorkspace,
+            originalFileName: policy?.originalFileName ?? policy?.id ?? report?.policyID,
+            isWorkspaceAvatar: true,
+        };
```

Comment:
Why do we have three cases? Can we change this to:
1. If report is related to policy use the old logic
2. Otherwise use report.avatarUrl